From 1e775bf2f43e9cb354e33c4acdcacd5c81779330 Mon Sep 17 00:00:00 2001
From: Heiko Schocher <hs@nabladev.com>
Date: Fri, 7 Jul 2023 09:34:59 +0200
Subject: [PATCH 13/23] ash: fix login problems with clutter

use for login the new function read_until_prompt_save
for the first commands tbot sets when entering linux shell

Signed-off-by: Heiko Schocher <hs@nabladev.com>
---
 tbot/machine/linux/ash.py | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/tbot/machine/linux/ash.py b/tbot/machine/linux/ash.py
index 90ffc1d..1828005 100644
--- a/tbot/machine/linux/ash.py
+++ b/tbot/machine/linux/ash.py
@@ -24,6 +24,8 @@ import tbot
 from .. import channel
 from . import linux_shell, util, special, path
 
+from tbot import log
+
 TBOT_PROMPT = b"TBOT-VEJPVC1QUk9NUFQK$ "
 
 Self = typing.TypeVar("Self", bound="Ash")
@@ -75,22 +77,22 @@ class Ash(linux_shell.LinuxShell):
                 + b"'",
             )
             self.ch.prompt = TBOT_PROMPT
-            self.ch.read_until_prompt()
+            self.ch.read_until_prompt_save()
 
             # Disable history
             self.ch.sendline("unset HISTFILE")
-            self.ch.read_until_prompt()
+            self.ch.read_until_prompt_save()
 
             # Disable line editing
             #
             # Not really possible on ash.  Instead, make the terminal really
             # wide and hope for the best ...
             self.ch.sendline("stty cols 1024")
-            self.ch.read_until_prompt()
+            self.ch.read_until_prompt_save()
 
             # Set secondary prompt to ""
             self.ch.sendline("PS2=''")
-            self.ch.read_until_prompt()
+            self.ch.read_until_prompt_save()
 
             # Do a sanity check to assert that shell interaction is working
             # exactly as expected
-- 
2.31.1

